File: cpp/1202.cpp
#include <iostream>
#include <stdio.h>
#include <string>

using namespace std;

int fibonacci(int n)
{
    int a = 0, b = 1, c = 0, i;

    if (n < 2) {
        return n;
    }

    for (i = 0; i < n - 1; i++) {
        c = a + b;
        c %= 1000;
        a = b;
        b = c;
    }

    return c;
}

int main()
{
    int t;
    cin >> t;

    for (t; t > 0; t--) {
        int n = 0, i;
        string binary;
        cin >> binary;

        for (i = 0; i < binary.length(); i++) {
            n *= 2;
            n += (int)(binary[i]) - 48;
            n %= 1500;
        }

        printf("%03d\n", fibonacci(n));
    }

    return 0;
}

File: py/1202.py
def fibonacci(x):
    a, b, c = 0, 1, 0

    if x < 2:
        return x

    else:
        for i in range(x - 1):
            c = a + b
            c %= 1000
            a = b
            b = c

        return c

t = int(input())

for i in range(t):
    n = 0
    binary  = input()

    for j in binary:
        n *= 2
        n += int(j)
        n %= 1500

    print(f'{(fibonacci(n)):03}')